modules/generator/processor/servicegraphs/servicegraphs.go

```diff
@@ -86,8 +86,20 @@ type Processor struct {
 
 func New(cfg Config, tenant string, registry registry.Registry, logger log.Logger) gen.Processor {
 	labels := []string{"client", "server", "connection_type"}
+
+	if cfg.EnableVirtualNodeLabel {
```

Comment:
why don't we add this to the list of labels above? this will prevent us from having to add lines 96-102 which adds special handling for this value.

then we could add a new param to `onComplete` where we pass the string from `onExpire`

Reply:
definitely doable - however what you save in this lines to avoid the prefixes, is added back to the onComplete logic for slice allocation and label value assignment.

also, it'll make this new label behave as half-dimension half-edge-property. do you think that'd be ok?

can you also explain further what's the modification of the onComplete signature for?

Reply:
yup, i see. i was trying to remove some of the state wrangling in the `consume()` method but you've already completely removed it.

agree that the tradeoffs mentioned:

> what you save in this lines to avoid the prefixes, is added back to the onComplete logic for slice allocation and label value assignment.

are not worth it. let's keep it as implemented now.

> also, it'll make this new label behave as half-dimension half-edge-property. do you think that'd be ok?

i'm not seeing an issue with that. do you have some concerns in mind?

Reply:
ok agree, leaving it behaving as before, as a dimension.

I think it's better to "treat it" as if it's fully one thing or the other. What I mean by that, from a future reader standpoint, it'll be easier to follow logic if it's treated as a dimension or as a struct property (first-class label?)